### PR TITLE
SI-7124 make macro definitions prettier in scaladoc

### PR DESCRIPTION
**Pull request by gourlaysama**
Macros are now prettier in scaladoc, by
- hiding the `macroImpl` annotation
- showing the `macro` modifier in front

---

review by @VladUreche

### CODE REVIEW

**Comment by soc** [comment]
Nice, LGTM.

File: src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala [hunk 1]
```diff
@@ -145,6 +146,7 @@ class ModelFactory(val global: Global, val settings: doc.Settings) {
        * any abstract terms, otherwise it would fail compilation. So we reset the DEFERRED flag. */
       if (!sym.isTrait && (sym hasFlag Flags.DEFERRED) && (!isImplicitlyInherited)) fgs += Paragraph(Text("abstract"))
       if (!sym.isModule && (sym hasFlag Flags.FINAL)) fgs += Paragraph(Text("final"))
+      if (sym.annotations.find(_.tpe =:= typeOf[macroImpl]).isDefined) fgs += Paragraph(Text("macro"))
```
Comment (xeno-by) [inline src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala]: `sym.isMacro` should be enough here

File: src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala [hunk 2]
```diff
@@ -80,7 +81,7 @@ class ModelFactory(val global: Global, val settings: doc.Settings) {
     def inTemplate: TemplateImpl = inTpl
     def toRoot: List[EntityImpl] = this :: inTpl.toRoot
     def qualifiedName = name
-    def annotations = sym.annotations.map(makeAnnotation)
+    def annotations = sym.annotations.filterNot(_.tpe =:= typeOf[macroImpl]).map(makeAnnotation)
```
Comment (xeno-by) [inline src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala:84]: Didn't know that scaladoc shows annotations. Does it?

Comment (gourlaysama) [inline src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala:84]: It does, when you expand to see the full desc., definition class, etc.  see the `@macroImpl` annotation in the taskKey method [in Sbt.Def](http://www.scala-sbt.org/0.13.1/api/#sbt.Def$).

File: src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala (hunk 1 above)
Comment (gourlaysama) [inline src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala]: Oh, that's much easier indeed. Updated.

File: src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala (hunk 2 above)
Comment (xeno-by) [inline src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala:84]: I see, thank you for the explanation!

**Comment by xeno-by** [comment]
LGTM

**Comment by VladUreche** [comment]
Post-merge LGTM from me too, sorry for being late to the party!

**Pull request merged**